kernel/src/callback.rs

```diff
@@ -6,41 +6,80 @@ pub struct AppId {
     idx: usize,
 }
 
+/// The kernel can masquerade as an app. IDs >= this value are the kernel.
+/// These IDs are used to identify which kernel container is being accessed.
+const KERNEL_APPID_BOUNDARY: usize = 100;
+
 impl AppId {
```

Comment:
I'm not sure I understand your discomfort here. The `is_kernel` and `is_kernel_idx` functions are very heavily tied to `AppId`s? If the were a global public function you would need to name them something like `is_value_in_kernel_appid_range(val: usize)` - it seems intrinsic to the AppId type that you'd ask it whether this `usize` is kernel or not?